### DISCUSSION TITLE
Rabbitmq normal cluster mode, accessed through VIP, the node goes offline and goes online again, and the original consumers disappear

### DISCUSSION THREAD

**Discussion by lemon-arrebol** [rabbitmq/rabbitmq-server#9528] [Other] [answered]
### Describe the bug

Rabbitmq normal cluster mode, accessed through VIP, the node goes offline and goes online again, and the original consumers disappear

### Reproduction steps

1、Rabbitmq Normal Cluster Deployment Mode
2、Keepalived+HAProxy load balancing high availability architecture
3、The cluster has 2 nodes, and node A already has several subscription queue consumers. Take node A offline first, and then take it online in 10 minutes. The consumer on node A has dropped the line

### Expected behavior

The consumer of node A exists and can continue to consume messages

**Comment by mkuratczyk** [comment]
It's not clear what you mean here. Of course the connections are dropped when the node goes down. It's the client's responsibility to reconnect. Please clarify.

Probably unrelated but 2-node clusters are a bad idea: https://www.rabbitmq.com/partitions.html#pause-minority, 
https://www.rabbitmq.com/quorum-queues.html#quorum-requirements

**Reply by lemon-arrebol** [reply]
Our rabbitmq deployment architecture is in VIP mode (Keepalived+HAProxy), followed by rabbitmq brokers. Clients access the cluster through VIP, and HAProxy forwards traffic to subsequent rabbitmq brokers

**Reply by lemon-arrebol** [reply]
I would like to confirm the mirror cluster mode. Will brokers within the cluster synchronize with consumers subscribing to the queue?

**Comment by lemon-arrebol** [comment]
The current environment is: two nodes are deployed in the same cluster, and RabbitMQ nodes after accessing the cluster through VIP (Keepalived+HAProxy). Take one of the nodes offline, and for the client, the connection should not be disconnected

**Reply by michaelklishin** [reply, marked as answer]
It's a responsibility of the client to reconnect. RabbitMQ does not connect to clients, clients connect (and reconnect as needed) to RabbitMQ. Some client libraries can do this [automatically if configured](https://rabbitmq.com/api-guide.html#recovery) and if you accept their recovery strategy as suitable.

Clients also share their part of responsibility for [using publisher](https://rabbitmq.com/publishers.html#data-safety) and [consumer-side](https://rabbitmq.com/confirms.html) data safety features.

**Comment by lemon-arrebol** [comment]
In normal cluster mode, will brokers within the cluster synchronize with consumers subscribing to the queue? If not synchronized, in VIP access mode: does it mean that after the node is offline, the original consumer channel will not be aware of the broker change, resulting in no consumers in the broker's view

**Reply by michaelklishin** [reply]
What do you mean by "syncrhonize with consumers"?

It's a responsibility of the client to reconnect. RabbitMQ does not connect to clients, clients connect (and reconnect as needed) to RabbitMQ. Some client libraries can do this [automatically if configured](https://rabbitmq.com/api-guide.html#recovery) and if you accept their recovery strategy as suitable.

Clients also share their part of responsibility for [using publisher](https://rabbitmq.com/publishers.html#data-safety) and [consumer-side](https://rabbitmq.com/confirms.html) data safety features.
